Add Nexus-Link header spec (#8)
nexus-rpc/api: SPEC.md

      Describes the current state of an operation.
```

> ### Links

Reviewer: I wouldn't put this in the schema section. These objects are never serialized in JSON form.

I would add another section to the doc called `Links` after [Predefined Handler Errors](https://github.com/nexus-rpc/api/blob/4d0ed524d8277e65823d2565974705958330d29e/SPEC.md#predefined-handler-errors)

Reviewer: In the new section you can borrow the text from MDN:

The Nexus-Link header field provides a means for serializing one or more links in HTTP headers. This header is encoded the same way as the HTTP header `Link` described
[here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Link).
Handlers and callers can specify links in different Nexus requests to associate an operation with arbitrary resources.

the handler. The callback request must strip away the `Nexus-Callback-` prefix. E.g if a Start Operation request
includes a `Nexus-Callback-Token: some-token` header, the callback request would include a `Token: some-token` header.

> The header `Nexus-Link` contains a `Link` object encoded the same way as the HTTP header `Link` described

Reviewer: You can shorten this and link to the `Links` header.

  **Body**: Arbitrary data conveying the operation's result. Headers should specify content type and encoding.

> - `201 Created`: Operation was started and will complete asynchronously. It may return `Nexus-Link` headers that
>   contain information about the operations done by the handler.

Reviewer: ```suggestion
- `201 Created`: Operation was started and will complete asynchronously. It may return `Nexus-Link` headers to
  associate resources to this operation.
```
